### PR TITLE
Improve the Fetch API

### PR DESCRIPTION
**Pull request by derekdreery**
I needed a fetch interface, and I didn't find any of the existing options exactly fit my needs (I needed access to some of the web-only options), so I implemented it in `gloo`.

The implementation tries to be as unopinionated as possible, just doing things like strongly typing results, and discarding impossible errors. I'm PRing it in case you're interested in having it upstream, but I'm also happy to just keep a private fork if you don't want it.

### CODE REVIEW

File: crates/fetch/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+    pub fn method(&mut self, method: &str) -> &mut Self {
+        self.init.borrow_mut().method(method);
+        self
+    }
+
+    /// Get access to the headers object for this request.
+    ///
+    /// If you are going to insert new headers, note that
+    /// [some names are forbidden](https://developer.mozilla.org/en-US/docs/Glossary/Forbidden_header_name)
+    /// and if they are set then sending the request will error.
```
Comment (derekdreery) [inline crates/fetch/src/lib.rs:49]: Need to check the spec to see what happens, it's possible the offending headers are silently discarded.

**Comment by derekdreery** [comment]
> Have you seen [reqwasm](https://docs.rs/reqwasm/latest/reqwasm/)? If not, please have a look. It tries to be exactly what you're PRing and then some.

Cool! your crate mostly superseeds my work, but there are a couple of points where I've done more (e.g. doc comments from MDN). It would be good to get `reqwasm` into `gloo` (possibly as `gloo::unstable::net` to start with, to give it time to stabilize). Then people can start hacking on it and improving it.

**Comment by hamza1311** [comment]
I have created https://github.com/rustwasm/gloo/pull/191

Once that is merged, I encourage you to update your branch to build off of it. One big improvement I can see is documentation: your docs are much better.

**Comment by derekdreery** [comment]
Yep happy to rebase off of your patch, which will make this PR mostly docs. :)

**Comment by hamza1311** [comment]
I have merged the PR. You can rebase now

**Comment by hamza1311** [comment]
@derekdreery any update on this? Are you working on the rebase?

**Comment by derekdreery** [comment]
Yes apologies I've been busy with work. Don't want to make promises but will look at ASAP.

**Comment by hamza1311** [comment]
I went ahead and merged master into this branch

**Comment by derekdreery** [comment]
Thanks for doing this - sorry I haven't had time.

**Pull request merged**